### PR TITLE
linux-testing: update to latest 5.13-rc6

### PR DESCRIPTION
**Pull request by ikervagyok**
###### Motivation for this change
`linux-testing` seems to be a bit stuck regardless of recent updates to most kernels.
~~This also seems to fix an `amdgpu` [bug](https://bugzilla.kernel.org/show_bug.cgi?id=212739) I see on a daily basis.~~

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/common-config.nix [hunk 1]
```diff
@@ -698,8 +698,8 @@ let
       THRUSTMASTER_FF    = yes;
       ZEROPLUS_FF        = yes;
 
-      MODULE_COMPRESS    = yes;
-      MODULE_COMPRESS_XZ = yes;
+      MODULE_COMPRESS    = whenOlder "5.13" yes;
+      MODULE_COMPRESS_XZ = whenOlder "5.13" yes;
```
Comment (xaverdh) [inline pkgs/os-specific/linux/kernel/common-config.nix:702]: why not keep the `MODULE_COMPRESS_XZ` / we still want to use xz compressed modules I think?

**Comment by xaverdh** [comment]
Not sure, but maybe the changes to `common-config` should be a separate commit (because its not just about the `rc` kernel). Also might benefit from some explanation / or reference to the upstream changes that make them necessary.

**Comment by ikervagyok** [comment]
Maybe the changes to the common config won't be necessary with later releases of 5.13.

**Comment by xaverdh** [comment]
Well it seems likely that they will still be needed.
Did you close this on purpose? Hope I didn't discourage you from contributing!

**Comment by ikervagyok** [comment]
@xaverdh you haven't discouraged me

I'll see if this is still necessary in a later release - maybe I was overeager to apply a fix from my machine to upstream nixpkgs. But you're right, I'll have to look into it for longer, instead of changing settings ad-hoc. I'll reopen with an updated release and separate commits in a week if/when I find the reason for the common config change in one of the the kernel patches for 5.13.

**Comment by ikervagyok** [comment]
@xaverdh found the reason for the changes to `common-config.nix`: 
removal of `/dev/kmem`:
    - https://cateee.net/lkddb/web-lkddb/DEVKMEM.html
    - https://github.com/torvalds/linux/commit/bbcd53c960713507ae764bf81970651b5577b95a
removal of `gma600`:
    - https://cateee.net/lkddb/web-lkddb/DRM_GMA600.html
removal of `module_compress`:
    - https://cateee.net/lkddb/web-lkddb/MODULE_COMPRESS.html
    - might want to enable one of these instead:
        - https://cateee.net/lkddb/web-lkddb/MODULE_COMPRESS_XZ.html - current default in nixpkgs
        - https://cateee.net/lkddb/web-lkddb/MODULE_COMPRESS_GZIP.html
        - https://cateee.net/lkddb/web-lkddb/MODULE_COMPRESS_NONE.html - maybe nice, since initrd is usualy compressed anyways
        - https://cateee.net/lkddb/web-lkddb/MODULE_COMPRESS_ZSTD.html - probably a good alternative

I'll test and push an updated branch after building is complete

File: pkgs/os-specific/linux/kernel/linux-testing.nix [hunk 2]
```diff
@@ -3,15 +3,15 @@
...
+  version = "5.13-rc6";
   extraMeta.branch = "5.12";
 
   # modDirVersion needs to be x.y.z, will always add .0
   modDirVersion = if (modDirVersionArg == null) then builtins.replaceStrings ["-"] [".0-"] version else modDirVersionArg;
 
   src = fetchurl {
     url = "https://git.kernel.org/torvalds/t/linux-${version}.tar.gz";
-    sha256 = "0w0zk2byimdbcvn8myqaq0ab6lyd43493fnkv9a1407dimpxb03d";
+    sha256 = "sha256-PunFd6tOsmrsPItp2QX4TEVxHnvvi1BMSwWio/DTlMU=";
```
Comment (xaverdh) [inline pkgs/os-specific/linux/kernel/linux-testing.nix:14]: Not sure, if we are allowed to use prefixed hashes in `nixpkgs` yet. @SuperSandro2000 might know.

**Review by xaverdh — COMMENTED** [review]
otherwise looks good to me!

File: pkgs/os-specific/linux/kernel/linux-testing.nix (hunk 2 above)
Comment (ikervagyok) [reply pkgs/os-specific/linux/kernel/linux-testing.nix:14]: I think it's OK:
```
$ grep -ri "sha256-" | wc -l
3503
```

Comment (SuperSandro2000) [reply pkgs/os-specific/linux/kernel/linux-testing.nix:14]: Greping through the code is not a valid way to determine if we should do something or not. If we would use that method all the time we would never get rid of old and deprecated patterns.

Using prefixed hashes is allowed and I have no problem with it.

**Pull request merged**